agent/src/beerocks/monitor/monitor_thread.cpp

```diff
@@ -346,6 +373,26 @@ void monitor_thread::after_select(bool timeout)
             }
         }
 
+        // Process nl events
+        if (mon_hal_nl_events) {
+            if (read_ready(mon_hal_nl_events)) {
+                if (!mon_wlan_hal->process_nl_events()) {
+                    LOG(ERROR) << "process_nl_events() failed!";
+                    thread_last_error_code = MONITOR_THREAD_ERROR_NL_REPORT_PROCESS_FAIL;
+                    stop_monitor_thread();
+                    return;
+                }
+                LOG(DEBUG) << "proccess nl events finished ";
+                clear_ready(mon_hal_nl_events);
+            }
+        } else {
+            // There is no socket for external events
+            LOG(ERROR) << "no socket exists for mon_hal_nl_events";
+            thread_last_error_code = MONITOR_THREAD_ERROR_NL_ATTACH_FAIL;
+            stop_monitor_thread();
+            return;
+        }
```

Comment:
Removed

Reply:
in case platform dosnt support netlink (dummy mode for example) we shouldn't call stop_monitor_thread() in cased FD is invalid.
@itayx  didn't you add a fixup that removes it ?

Reply:
yeah, I commented saying I did remove it

Reply:
was not clear :)